src/device-manager/java/src/nl/Weave/DataManagement/WdmClient.java

```diff
@@ -67,11 +67,22 @@
      */
     public void beginRefreshData();
 
+    /**
+     * The result of this operation can be observed through the {@link CompletionHandler}
+     * that has been assigned via {@link #setCompletionHandler}.
+     * The fetched events can be accessed via getEvents()
+     *
+     * @param timeout operation timeout, this operation might take a long time if the number of events is too large.
+     */
+    public void beginFetchEvents(int timeout);
+
     public CompletionHandler getCompletionHandler();
     public void setCompletionHandler(CompletionHandler compHandler);
 
     public GenericTraitUpdatableDataSink getDataSink(long traitInstancePtr);
 
+    public String getEvents();
+
     public interface CompletionHandler
```

Comment:
Do we need a new CompletionHandler interface method now for onFetchEventsComplete()? Ideally that would just provide the String directly, which could avoid the ambiguity of the getEvents() return value.

Reply:
Adds a new function `onFetchEventsComplete` in `WdmClient.CompletionHandler` interface, however, I would prefer to preserve getEvents() so we can test Java binding more easily.
Currently, onFetchEventsComplete is just a warpper of `onXxxxComplete` and `getEvents`